src/jimp2/wireworld/z8/datamangment/DataManager.java

```diff
@@ -19,19 +28,77 @@ public void writeIterationToFile(int iteration, World world, List<Element> eleme
         // TODO implement here
     }
 
-    private Element interpretInputPiece(String input) {
-        // TODO implement here
+    private Element interpretInputPiece(JSONObject jsonObject, String name) {
+        int x1,x2,y1,y2;
+        int width, height;
+        Point point;
+        switch (name) {
+            case "Wire":
+                x1 = ((Long) jsonObject.get("x1")).intValue();
+                x2 = ((Long) jsonObject.get("x2")).intValue();
+                y1 = ((Long) jsonObject.get("y1")).intValue();
+                y2 = ((Long) jsonObject.get("y2")).intValue();
+                Point posiotion1 = new Point(x1, y1);
+                Point posiotion2 = new Point(x2, y2);
+                return new Wire(posiotion1, posiotion2);
+            case "Electron":
+                x1 = ((Long) jsonObject.get("x")).intValue();
+                y1 = ((Long) jsonObject.get("y")).intValue();
+                point = new Point(x1, y1);
+                String orientation = (String) jsonObject.get("orientation");
+                Orientation orient = null;
+                for (Orientation o : Orientation.values()) {
+                    if (o.toString().equals(orientation))
+                        orient = o;
+                }
+                if (orient == null)
+                    System.err.println("Wrong orientation in data files (Electron)" + point.toString());
+                return new Electron(point, orient);
+            case "Generator":
+                    x1 = ((Long) jsonObject.get("x")).intValue();
+                    y1 = ((Long) jsonObject.get("y")).intValue();
+                    width = ((Long) jsonObject.get("width")).intValue();
+                    height = ((Long) jsonObject.get("width")).intValue();
+                    point = new Point(x1, y1);
+                    return new Generator(point, width, height);
+        }
         return null;
     }
 
     public WorldData readInputFile() {
-        // TODO implement here
+        inputFile = new File("data.json");
```

Comment:
Potrzebujemy możliwości wybrania pliku - powinno korzystać z jakiejś globalnej zmiennej, którą określa metoda setInputFile()

Reply:
tak wiem, to tymczasowe rozwiązanie